cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java

```diff
@@ -942,4 +952,73 @@ private static String trimStartingWhitespaces(String str) {
     private static String removeLastCharacter(String str) {
         return str.substring(0, str.length() - 1);
     }
+
+    /**
+     * Get the latest version from a given list of versions.
+     *
+     * @param versions the list of strings
+     * @return the latest version
+     */
+    static String getLatestVersion(List<String> versions) {
+        String latestVersion = versions.get(0);
+        for (String version : versions) {
+            if (SemanticVersion.from(version).greaterThan(SemanticVersion.from(latestVersion))) {
+                latestVersion = version;
+            }
+        }
+        return latestVersion;
+    }
+
+    /**
+     * Pull the dependencies of a given package from central.
+     *
+     * @param orgName org name of the dependent package
+     * @param packageName name of the dependent package
+     * @param version version of the dependent package
+     * @return true if the dependent package compilation has errors
+     */
+    static boolean pullDependencyPackages(String orgName, String packageName, String version) {
+        Path ballerinaUserHomeDirPath = ProjectUtils.createAndGetHomeReposPath();
+        Path centralRepositoryDirPath = ballerinaUserHomeDirPath.resolve(ProjectConstants.REPOSITORIES_DIR)
+                .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
+        Path balaDirPath = centralRepositoryDirPath.resolve(ProjectConstants.BALA_DIR_NAME);
+        Path balaPath = ProjectUtils.getPackagePath(balaDirPath, orgName, packageName, version);
+        String ballerinaShortVersion = RepoUtils.getBallerinaShortVersion();
+        Path cacheDir = centralRepositoryDirPath.resolve(
+                ProjectConstants.CACHES_DIR_NAME + "-" + ballerinaShortVersion);
+
+        ProjectEnvironmentBuilder defaultBuilder = ProjectEnvironmentBuilder.getDefaultBuilder();
+        defaultBuilder.addCompilationCacheFactory(new FileSystemCache.FileSystemCacheFactory(cacheDir));
+        BalaProject balaProject = BalaProject.loadProject(defaultBuilder, balaPath);
+
+        // Delete package cache if available
+        Path packageCacheDir = cacheDir.resolve(orgName).resolve(packageName).resolve(version);
+        if (packageCacheDir.toFile().exists()) {
+            deleteDirectory(packageCacheDir);
+        }
+
+        // getResolution pulls all dependencies of the pulled package
+        PackageCompilation packageCompilation = balaProject.currentPackage().getCompilation();
+        Collection<Diagnostic> resolutionDiagnostics = packageCompilation.getResolution()
```

Comment:
we can break the logic from this line onwards to a separate method for readability.
`packageCompilation` as an argument and method will have the purpose of checking whether there are any error diagnostics. WDYT?